arch/posix/CMakeLists.txt

```diff
@@ -1,5 +1,12 @@
 # SPDX-License-Identifier: Apache-2.0
 
+execute_process(COMMAND arch OUTPUT_VARIABLE BUILD_ARCH OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+if (${BUILD_ARCH} STREQUAL "aarch64")
+	zephyr_compile_options(-fPIC)
```

Comment:
(Disclaimer I do not have any experience compiling in an ARM environment)
* Maybe a stupid question, but, do we know in general if the ABI defaults to lp64 in aarch64? (and/or would it need to be specified explicitly?)
* I guess that with this code as is only native_posix_64 compiles as expected?
  * And if that is the case I guess a `if (CONFIG_64BIT)` before this line and an else below with an error saying something like "only native_posix_64 is supported so far in aarch64" would be needed.
  * Or a change so that `-mabi=ilp32` / `-mabi=lp64` is selected depending on `CONFIG_64BIT` (and whatever else would be needed)

Reply:
It's a bit tricky. Since Aarch64 is not multilib capable, it doesn't really make sense to specify the abi for the native compiler.
However, `-fPIC` does work regardless of whether the system is running in 32-bit mode or 64-bit mode and it is required for 64-bit mode.

Reply:
Uhm..
One clarification: CONFIG_64BIT is set when you select the target (board) you build for. So if you `-DBOARD=native_posix_64` you are asking to build targeting native_posix_64 which sets CONFIG_64BIT (just like when you select a RISCV-64 based board. Note that native_posix* is the only not-crosscompiled target)
If you target native_posix you are explicitly asking for the 32 bit version.
So, in that regard, if somebody asks for native_posix_64 and you decide for them to build the 32bit version (because that's the only user space they have) you would be "cheating", and probably confusing them. In that case, and if you wanted to be extra nice, I would just error out with a nice explanation ("You can't build native_posix_64 in a 32bit OS")

> Option 1:
> CONFIG_64BIT is not set by default on y machine, and the build fails (as it should?)

=> The fault in this case is in selecting native_posix(_32).

> Option 2:
> Aarch64 requires -fPIC but it also does not hurt to use it for Aarch32. Use it as a blanket option?

That should be fine. I do not have any quarrel with setting this option always :)

> I guess the underlying question is: is it "bad" to run native_posix tests on aarch64 without caring if it is 32-bit or 64-bit?

Not worse than in x86: It should just not work: if you try to build for one of them, it won't if you do not have the libC for that one (32/64), or fail to run later if the loader cannot handle it.

Reply:
Note that the only proper way to support both native_posix(_32) and native_posix_64 is having multilib support in your host. If you don't, then you can only use one. And that is probably just fine for whoever is trying to build native_posix in ARM.